src/matchers/numbers.test.lua

```diff
@@ -0,0 +1,653 @@
+
+				-- Then
+				assertMatcher.passed(resultEqual)
+			end
+		)
+
+		describe("should fail", function()
+			test(
+				"when the received value is greater than the expected value",
+				function()
+					-- Given
+					local greaterNumber = 10
+					local smallerNumber = 3
+
+					-- When
+					local result = matchers.toBeLessThanOrEqual(
+						CONTEXT,
+						greaterNumber,
+						smallerNumber
+					)
+
+					-- Then
+					assertMatcher.failed(result)
+					assertMatcher.hasMessage(
+						result,
+						("Expected %d to be less than or equal to %d"):format(
+							greaterNumber,
+							smallerNumber
+						)
+					)
+				end
+			)
+
+			test("on NaN for any parameter", function()
+				-- Given
+				local normalNumber = 1
+
+				-- When
+				local result =
+					matchers.toBeLessThanOrEqual(CONTEXT, normalNumber, NAN)
+
+				local resultSwapped =
+					matchers.toBeLessThanOrEqual(CONTEXT, NAN, normalNumber)
+
+				-- Then
+				assertMatcher.failed(result)
+				assertMatcher.hasMessage(
+					result,
+					("Expected %d to be less than or equal to %s"):format(
+						normalNumber,
+						tostring(NAN)
+					)
+				)
+				assertMatcher.failed(resultSwapped)
+				assertMatcher.hasMessage(
+					resultSwapped,
+					("Expected %s to be less than or equal to %d"):format(
+						tostring(NAN),
+						normalNumber
+					)
+				)
+			end)
+		end)
+
+		it("should have an inverted message", function()
+			-- Given
+			local greaterNumber = 10
+			local smallerNumber = 2
+			-- When
+			local result = matchers.toBeLessThanOrEqual(
+				INVERTED_CONTEXT,
+				greaterNumber,
+				smallerNumber
+			)
+
+			-- Then
+			assertMatcher.hasMessage(
+				result,
+				("Expected %d to not be less than or equal to %d"):format(
+					greaterNumber,
+					smallerNumber
+				)
+			)
+		end)
+	end)
+
+	describe("toBeNaN", function()
+		it("should pass when the received value is NaN", function()
+			-- Given
+			local notANumber = 0 / 0
+
+			-- When
+			local result = matchers.toBeNaN(CONTEXT, notANumber)
+
+			-- Then
+			assertMatcher.passed(result)
+		end)
+
+		it("should fail when the received value isn't NaN", function()
+			-- Given
+			local normalNumber = 1
+
+			-- When
+			local result = matchers.toBeNaN(CONTEXT, normalNumber)
+
+			-- Then
+			assertMatcher.failed(result)
+			assertMatcher.hasMessage(
+				result,
+				("Expected %d to be NaN"):format(normalNumber)
+			)
+		end)
+
+		it("should have an inverted message", function()
+			-- Given
+			local notANumber = 0 / 0
+
+			-- When
+			local result = matchers.toBeNaN(INVERTED_CONTEXT, notANumber)
+
+			-- Then
+			assertMatcher.hasMessage(
+				result,
+				("Expected %s to not be NaN"):format(tostring(notANumber))
+			)
+		end)
+	end)
+
+	describe("toBeInfinity", function()
+		it("should pass when the received value is infinity", function()
+			-- Given
+			local inf = math.huge
+
+			-- When
+			local result = matchers.toBeInfinity(CONTEXT, inf)
+
+			-- Then
+			assertMatcher.passed(result)
```

Comment:
Please also test negative infinity here.

Also, there's no need for the arrange step, you can just use `math.huge` directly.
On a more general note, don't try to always have an arrange, act, and assert in every test. For most tests this will be the natural pattern, but there are cases (like this) where skipping one or even two of the sections is perfectly valid.